61-matamata-turtle/technical_specification.md

```diff
@@ -0,0 +1,300 @@
+# DLQ Service (Matamata Turtle)
+**Epic Type:** Implementation Epic
+
+Epic planning and implementation follow the
+[Epic Planning and Marathon SOP](https://docs.ghga-dev.de/main/sops/sop001_epic_planning.html).
+
+## Scope
+### Outline:
+This epic is concerned with the implementation of a basic dead letter queue service,
+which will provide the means to assess, process, and retry Kafka events that resulted
+in errors during their initial processing. This service will make use of the
+[Kafka DLQ](https://github.com/ghga-de/hexkit/pull/123) additions to `hexkit`.
+
+### Included/Required:
+- Implementation of the DLQ service (including its test suite)
+- `hexkit`: Add exception information to DLQ event headers
+- `hexkit`: Add preview functionality to `KafkaDLQSubscriber`
+- `hexkit`: Add preview dry-run functionality to `KafkaDLQSubscriber`
+
+### Optional:
+- Add event ID header in `hexkit`, consisting of the original topic, partition, and
+offset
+
+### Not included:
+- A web-based user interface
+
+## API Definitions:
+
+### RESTful/Synchronous:
+
+DLQ topics consist of the original topic name, the service abbreviation, and the fixed
+suffix `-dlq`. The latter could technically be omitted, but using it visually
+distinguishes dlq topics from others and making debugging easier.
+When interacting with DLQ topics via the API, however, the above structure does not need
+to be remembered: only the service abbreviation and the plain topic name.
+
+- `GET /services`
+  - *Returns a list of all configured services*
+  - Auth Header: internal token
+  - Response Body: array of strings (service names)
+  - Response Status: `200 OK`
+- `GET /topics/{service}`
+  - *Returns a list of all topics configured for the chosen service*
+  - Auth Header: internal token
+  - Response Body: array of strings (topic names)
+  - Response Status: `200 OK`
+- `GET /{service}/{topic}`
```

Comment:
Should we add query params here to allow pagination (limit and offset)?

Reply:
I've made the change